src/api/ans.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Account } from "../core";
+import { getOwnerAddress, registerDomain } from "../internal/ans";
+import { SingleSignerTransaction, GenerateTransactionOptions } from "../transactions/types";
+import { HexInput, MoveAddressType } from "../types";
+import { AptosConfig } from "./aptos_config";
+
+/**
+ * A class to handle all `ANS` operations
+ */
+export class ANS {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  /**
+   * Retrieve the owner address of a domain name or subdomain name.
+   *
+   * ```ts
+   * getOwnerAddress({domainName: "aptos", subdomainName: "test"})
+   * // Will return the owner address of "test.aptos.apt" or undefined
+   * ```
+   *
+   * @param args
+   * @param args.domainName - A string of the domain name to retrieve
+   * @param args.subdomainName - A string of the subdomain name to retrieve
+   *
+   * @returns MoveAddressType if the name is owned, undefined otherwise
+   */
+  async getOwnerAddress(args: { domainName: string; subdomainName?: string }): Promise<MoveAddressType | undefined> {
+    return getOwnerAddress({ ...args, aptosConfig: this.config });
```

Comment:
dont think it makes any difference, but it is a standard
```suggestion
    return getOwnerAddress({ aptosConfig: this.config, ...args });
```